File: python/leetcode014.py
class Solution():
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """

        ls = len(strs)
        if ls == 0:
            return ''
        res = strs[0]
        for i in range(1,ls):
            res = self.compare(res, strs[i])
        return res


    def compare(self, res, str):
        m = len(res)
        n = len(str)
        result = ''
        i =0
        while i < m and i < n:
            if res[i] == str[i]:
                result = result + res[i]
                i = i + 1
            else:
                break
        return result

if __name__ == "__main__":
    s = Solution()
    print(s.longestCommonPrefix(["flower","flow","flight"]))

File: leetcodec++/leetcode014.cpp
//
// Created by 黄敏辉 on 2019-04-26.
//

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.size() == 0) return "";
        string s = strs[0];
        for (int i = 0; i < strs.size(); i++){
            while (strs[i].find(s) != 0){
                s = s.substr(0, s.length()-1);
                if (s.length() == 0) return "";
            }
        }
        return s;
    }
};